internal/cli/create_ext.go

```diff
@@ -0,0 +1,112 @@
+package cli
+
+import (
+	"context"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/tilt-dev/tilt/internal/analytics"
+	engineanalytics "github.com/tilt-dev/tilt/internal/engine/analytics"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+// A human-friendly CLI for creating extensions.
+type createExtCmd struct {
+	helper *createHelper
+	cmd    *cobra.Command
+
+	repoName string
+	repoPath string
+}
+
+var _ tiltCmd = &createExtCmd{}
+
+func newCreateExtCmd() *createExtCmd {
+	helper := newCreateHelper()
+	return &createExtCmd{
+		helper: helper,
+	}
+}
+
+func (c *createExtCmd) name() model.TiltSubcommand { return "create" }
+
+func (c *createExtCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "ext NAME URL [ARG...]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Register an extension.",
+		Long: `Register an extension with a running Tilt instance.
+
+An extension will load a set of services into your dev environment.
+
+These might be services you need to run your app, or servers
+that add functionality to Tilt itself.
+
+Assumes that an extension repo has already been registered
+with 'tilt create repo' or in the Tiltfile.
+`,
+		Args: cobra.MinimumNArgs(1),
+		Example: `
+# Installs the extension from the extension repo 'default' under the path './cancel'.
```

Comment:
Pedantic: isn't this actually `$XDG_DATA_HOME/tilt-dev/cancel`?

(Only mentioning because it's not immediately clear to me if `path` is just meant to rename the subdirectory or to allow you to re-root the extension entirely to e.g. `/foo/bar/cancel` outside of XDG data home)

Reply:
it's actually more like tilt-dev/github.com/tilt-dev/tilt-extensions/cancel (since it's relative to the repo root)